transformations/slangificator/transformation.py

```diff
@@ -0,0 +1,271 @@
+import itertools
+import random
+import spacy
+import os
+
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+def slangify(
+    self,
+    text,
+    probReplaceNoun=0.5,
+    probReplaceAdjective=0.5,
+    probReplaceAdverb=0.5,
+    seed=0,
+    max_outputs=1,
+):
+    random.seed(seed)
+
+    nlp = spacy.load("en_core_web_sm")  # get an instnace of the tikenizer
+    perturbed_texts = []  # output for all perturbed texts
+
+    # Load dictionaries
+    Slang_Nouns = self.Slang_Nouns
+    Slang_Nouns = list(map(list, zip(*Slang_Nouns)))
+    Slang_Adverbs = self.Slang_Adverbs
+    Slang_Adverbs = list(map(list, zip(*Slang_Adverbs)))
+    Slang_Adjectives = self.Slang_Adjectives
+    Slang_Adjectives = list(map(list, zip(*Slang_Adjectives)))
+
+    # Tags for nouns
+    noun_tag = ["NN", "NNS", "NNPS", "NNP"]
+
+    # Tokenize text
+    doc = nlp(text)
+
+    for _ in itertools.repeat(None, max_outputs):
+
+        ReplPot = 0  # counts potential replcacements, which could have been made if the probability of replacement would be set to one for all PoS
+        ReplMade = 0  # counts actual replcacements
+
+        modified_toks = []  # modified tokens
+        for token in doc:
+            isCap = token.text[
+                0
+            ].isupper()  # check if token begins with the capital letter
+
+            # Nouns
+            if token.tag_ in noun_tag:
```

Comment:
If you could split the nouns, adverbs and adjectives functionality into one common function - Since most of the logic is the same except for the POS tags - you could pass those as a variable in the function itself. And call it thrice with different POS types passed each time.

Reply:
@kaustubhdhole thanks for the proposal. I think @sebastianGehrmann  was pointing to a very similar issue but that time I have not realised completely how to make the logic more abstract.
So in the most recent version, I added a new function, which absorbs all of the similar logic for the transformation

Reply:
This looks neater! Thanks for making the change!